File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/client/docusaurus/Optimizations.java
package io.getstream.chat.docs.java.client.docusaurus;

import static io.getstream.result.call.CallKt.forceNewRequest;

import android.content.Context;

import java.util.List;

import io.getstream.chat.android.client.ChatClient;
import io.getstream.chat.android.client.api.models.QueryChannelsRequest;
import io.getstream.result.call.Call;
import io.getstream.chat.android.models.Channel;

/**
 * @see <a href="https://getstream.io/chat/docs/sdk/android/client/guides/optimizations/">Optimizations</a>
 */
public class Optimizations {

    public void multipleCallsToAPI(Context context, String apiKey, QueryChannelsRequest queryChannelsRequest) {
        new ChatClient.Builder(apiKey, context).disableDistinctApiCalls().build();

        Call<List<Channel>> queryChannelsCall = ChatClient.instance().queryChannels(queryChannelsRequest);
        forceNewRequest(queryChannelsCall);
        queryChannelsCall.enqueue(result -> {
            if (result.isSuccess()) {
                // Handle success
            } else {
                // Handle error
            }
        });
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/client/docusaurus/Optimizations.kt
package io.getstream.chat.docs.kotlin.client.docusaurus

import android.content.Context
import io.getstream.chat.android.client.ChatClient
import io.getstream.chat.android.client.api.models.QueryChannelsRequest
import io.getstream.chat.android.client.extensions.forceNewRequest
import io.getstream.result.Result

/**
 * @see <a href="https://getstream.io/chat/docs/sdk/android/client/guides/optimizations/">Optimizations</a>
 */
class Optimizations {

    fun multipleCallsToAPI(context: Context, apiKey: String, queryChannelsRequest: QueryChannelsRequest) {
        ChatClient.Builder(apiKey, context).disableDistinctApiCalls().build()

        ChatClient.instance().queryChannels(queryChannelsRequest).forceNewRequest().enqueue { result ->
            when (result) {
                is Result.Success -> {
                    // Handle success
                }
                is Result.Failure -> {
                    // Handle error
                }
            }
        }
    }
}
